### ISSUE TITLE
Disable Conv2D bias when followed by a batch norm

### ISSUE THREAD

**Issue by jder** [m2lines/Samudra#104]
From the [PyTorch tuning guide](https://pytorch.org/tutorials/recipes/recipes/tuning_guide.html#disable-bias-for-convolutions-directly-followed-by-a-batch-norm):

> Disable bias for convolutions directly followed by a batch norm [torch.nn.Conv2d()](https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html#torch.nn.Conv2d) has bias parameter which defaults to True (the same is true for [Conv1d](https://pytorch.org/docs/stable/generated/torch.nn.Conv1d.html#torch.nn.Conv1d) and [Conv3d](https://pytorch.org/docs/stable/generated/torch.nn.Conv3d.html#torch.nn.Conv3d) ).
> 
> If a nn.Conv2d layer is directly followed by a nn.BatchNorm2d layer, then the bias in the convolution is not needed, instead use nn.Conv2d(..., bias=False, ....). Bias is not needed because in the first step BatchNorm subtracts the mean, which effectively cancels out the effect of bias.

**Comment by suryadheeshjith** [comment]
We are moving away from batch normalization layers to other forms of normalization like layer or instance norm. Would the same apply in these cases?

**Comment by suryadheeshjith** [comment]
Instance Normalization by default has affine=False which is what we use, but for layer normalization it looks like it would depend on the dimensions we normalize over (since we have elementwise affine).

**Comment by jder** [comment]
Yeah, probably depends on the details of the norm; maybe we can disable it depending on chosen norm.
